lib/ui/window.dart

```diff
@@ -294,23 +357,31 @@ class Locale {
       return false;
     final Locale typedOther = other;
     return languageCode == typedOther.languageCode
-        && countryCode == typedOther.countryCode;
+        && scriptCode == typedOther.scriptCode
+        && countryCode == typedOther.countryCode
+        && _variants == typedOther._variants;
   }
 
   @override
   int get hashCode {
     int result = 373;
     result = 37 * result + languageCode.hashCode;
-    if (_countryCode != null)
-      result = 37 * result + countryCode.hashCode;
+    result = 37 * result + scriptCode.hashCode;
+    result = 37 * result + countryCode.hashCode;
+    result = 37 * result + _variants.hashCode;
     return result;
```

Comment:
we should use hashValues/hashList as in the code snippet I pasted recently.

Reply:
Done, thanks!